crates/jstz_engine/src/value/conversions/try_from_js.rs

```diff
@@ -0,0 +1,129 @@
+use crate::{
+    context::{CanAccess, CanAlloc, Context, InCompartment},
+    gc::{ptr::AsRawPtr, Compartment},
+    letroot,
+    string::JsString,
+    value::JsValue,
+};
+
+/// This trait adds a fallible and efficient conversions from a [`JsValue`] to Rust types.
+///
+/// Note that the Output type is not directly linked to
+#[allow(unused)]
+pub trait TryFromJs<'a, 'cx: 'a, C: Compartment> {
+    type Output;
+    type Config;
+
+    /// This function tries to convert a JavaScript value into `Self::Output`.
+    /// `config` determines the behaviour of the conversion.
+    fn try_from_js<S>(
+        value: &JsValue<'a, C>,
+        config: Self::Config,
+        cx: &'cx mut Context<S>,
+    ) -> Option<Self::Output>
+    where
+        S: InCompartment<C> + CanAccess + CanAlloc;
+}
+
+impl<'a, 'cx: 'a, C: Compartment> TryFromJs<'a, 'cx, C> for i32 {
+    type Output = i32;
+
+    type Config = ();
```

Comment:
mozjs permits us to have more than 1 kind of conversion behaviour for `i32`s (and other int types) -- namely:
underflows/overflows, enforcing range (throwing an exception if not in range), and clamping. 

We should probably support these behaviours (since there are many other ways to obtain an `i32` from `JsValue` than just the `Number` variant in `JsValue`)